packages/server/src/utilities/redis.ts

```diff
@@ -4,23 +4,33 @@ import { ContextUser } from "@budibase/types"
 
 const APP_DEV_LOCK_SECONDS = 600
 const AUTOMATION_TEST_FLAG_SECONDS = 60
-let devAppClient: any, debounceClient: any, flagClient: any
+let devAppClient: any, debounceClient: any, flagClient: any, socketClient: any
 
-// we init this as we want to keep the connection open all the time
+// We need to maintain a duplicate client for socket.io pub/sub
+let socketSubClient: any
```

Comment:
The socket.io redis adapter requires two separate redis clients, one for pub and one for sub. I can't use another instance of our redis wrapper class for this, as they reuse the same clients depending on the "selectDb" flag. The easiest way to handle this was simply maintain a duplicate of core redis client here, and ensure it's also cleaned up when appropriate.